monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -545,6 +573,44 @@ sealed abstract class Task[+A] extends Serializable {
       def onError(ex: Throwable): Unit = f(Failure(ex))
     })(s)
 
+  /** Memoizes (caches) the result of the source task and reuses it on
+    * subsequent invocations of `runAsync`.
+    *
+    * The resulting task will be idempotent, meaning that
+    * evaluating the resulting task multiple times will have the
+    * same effect as evaluating it once.
+    *
+    * $unsafeMemoize
+    *
+    * @see [[memoizeOnSuccess]] for a version that only caches
+    *     successful results
+    *
+    * @return a `Task` that can be used to wait for the memoized
+    *         value or to cancel the background process that calculates
+    *         the memoized result (as long as it can be cancelled)
+    */
+  final def memoize: Fiber[A] =
```

Comment:
Excuse me if I'm totally wrong here

I don't really see how `Fiber` is a way to represent memoization. My intuition tells me that when I get a `Fiber[A]`, the evaluation process is already running, so my options are either waiting for its completion or telling it that it's no longer necessary (which might still continue running, but I'll never get a result). But implementation of `TaskMemoize` seems to imply the evaluation will only happen when I'll try to perform a `join`, so it doesn't really represent a _running background process_.

Reply:
You're totally right, haven't thought this through. I'm reverting the change.